include/arch/arm/cortex_m/error.h

```diff
@@ -38,6 +38,11 @@ extern void _SysFatalErrorHandler(unsigned int reason, const NANO_ESF *esf);
 #if defined(CONFIG_ARMV6_M)
 /* ARMv6 will hard-fault if SVC is called with interrupts locked. Just
  * force them unlocked, the thread is in an undefined state anyway
+ *
+ * On ARMv7 we won't get a hardfault, but if interrupts were locked the
```

Comment:
Implementation looks ok, but commit message needs to be updated. This affects ARM-v7m, which is M3, M4, and M7.